linux_files/setup

```diff
@@ -89,44 +76,89 @@ function byemessage {
+  "1 EXPLORER" "Enable right-click on folders in Windows Explorer to open them in WLinux" \
+  "2 SHELLS" "Install and configure default shells from bash, zsh, csh, and fish" \
+  "3 EDITORS" "Install text editors neovim, emacs, or Visual Studio Code" \
+  "4 PYTHON" "Install Python 3.7 and download and install latest PIP" \
+  "5 NODEJS"  "Install Node.js" \
+  "6 GO" "Install the latest Go" \
+  "7 POWERSHELL and AZURE" "Install PowerShell for Linux and Azure CLI Tools" \
+  "8 BASIC GUI TOOLS" "Install the basic libraries needed for most GUI apps and optionally set HiDPI" \
+  "q QUIT" "Quit This Menu Program"  3>&1 1>&2 2>&3)
+  
+  RET=$?
+  if [ $RET -eq 1 ]; then
+    exit 0
+  elif [ $RET -eq 0 ]; then
+    case "$SELECTION" in
+      1\ *) explorerintegration
+            mainmenu ;;
+      2\ *) installandsetshell
+            fzfinstall
+            mainmenu ;;
+      3\ *) neoviminstall
+            emacsinstall
+            codeinstall
+            mainmenu ;;
+      4\ *) pythoninstall
+            mainmenu ;;
+      5\ *) nodeinstall
+            mainmenu ;;
+      6\ *) goinstall
+            mainmenu ;;
+      7\ *) powershellinstall
+            azurecliinstall
+            mainmenu ;;
+      8\ *) basicguiinstall
+            hidpiconfig
+            mainmenu ;;
+      q\ *) echo "Exiting menu..."
+         *) whiptail --msgbox "Unrecognized option" 20 60 1 ;;
+    esac || whiptail --msgbox "There was an error running selection $SELECTION" 20 60 1
+  fi
+}
+
 ## internationalization
 
 function languageprompt {
-read -r -p "Would you like to configure additional languages? [y/N] " response
-if [[ "$response" =~ ^[yY]$ ]]
+if (whiptail --title "International?" --yesno "Would you like to configure additional languages in WLinux?" 8 65) then
 then
     echo "Running $ dpkg-reconfigure locales"
     sudo dpkg-reconfigure locales
 fi
 }
 
-## package meta functions
-
-    # list all package sets and the order they should be presented for optional install here
-
-function installpackages {
-echo "Shells:"
-installandsetshell
-echo "Editors:"
-neoviminstall
-emacsinstall
-echo "Enhancements:"
-fzfinstall
-explorerintegration
-## package set functions
-echo "Development Environment:"
-pythoninstall
-nodeinstall
-goinstall
-powershellinstall
-echo "Cloud Tools:"
-azurecliinstall
-echo "GUI Applications:"
-basicguiinstall
-codeinstall
-chromeinstall
-hidpiconfig
-}
+# package index
+
+# shells
+# installandsetshell
+
+# editors
+# neoviminstall
+# emacsinstall
+# codeinstall
+
+# enhancements
+# explorerintegration
+# fzfinstall
+# addrepos
+
+# development environments
+# pythoninstall
+# nodeinstall
+# goinstall
+
+# cloud tools
+# azurecliinstall
+# powershellinstall
+
+# gui tools
+# basicguiinstall
+# hidpiconfig
+
 
 function updateupgrade {
 echo "Updating apt package index from repositories: $ sudo apt update"
```

Comment:
using `whiptail` for this maybe?

Reply:
The plan is to slowly convert more over to whiptail, to the point will be there entirely.